linesearch.go

```diff
@@ -71,29 +75,30 @@ func (l *Linesearch) Iterate(loc *Location, xNext []float64) (EvaluationType, It
 	}
 	if l.Method.Finished(linesearchLocation) {
 		if l.lastEvalType == FunctionEval {
-			l.finished = true
-			l.finishedF = loc.F
 			// We have the function value at the current location, but we don't
-			// have the gradient, so get it before starting the next major iteration.
+			// have the gradient, so get it before announcing MajorIteration.
+			l.nextIterType = SubIteration
 			copy(xNext, loc.X)
-			return GradientEval, SubIteration, nil
+			return GradientEval, l.nextIterType, nil
 		}
-		return l.initNextLinesearch(loc, xNext)
+		// No gradient evaluation is necessary, announce MajorIteration.
```

Comment:
// The linesearch is finished. Announce so with an update to MajorIteration. The function value and gradient is already known, so no function evaluations are necessary.